### ISSUE TITLE
Test failures on Win10 with PowerShell and Strawberry

### ISSUE THREAD

**Issue by genio** [perl-libwin32/win32#22]
Using Strawberry Perl (Portables) on Windows 10, I'm having trouble installing the latest Win32 distribution via PowerShell. Please let me know if there is any other information I can provide to help debug:

```log
PS C:\Users\cwhitener> cpanm -v --test-only Win32
cpanm (App::cpanminus) 1.7044 on perl 5.030000 built for MSWin32-x86-multi-thread-64int
Work directory is C:\Users\cwhitener\_psperl\_perls\perl32w64int-5.30.0.1\data/.cpanm/work/1584363053.9400
You have make C:\Users\cwhitener\_psperl\_perls\perl32w64int-5.30.0.1\c\bin\gmake.exe
You have LWP 6.39
Falling back to Archive::Tar 2.36
Searching Win32 () on cpanmetadb ...
--> Working on Win32
Fetching http://www.cpan.org/authors/id/J/JD/JDB/Win32-0.53.tar.gz ... OK
Unpacking Win32-0.53.tar.gz
Entering Win32-0.53
Checking configure dependencies from META.yml
Running Makefile.PL
Configuring Win32-0.53 ... Checking if your kit is complete...
Looks good
Generating a gmake-style Makefile
Writing Makefile for Win32
Writing MYMETA.yml and MYMETA.json
OK
Checking dependencies from MYMETA.json ...
Checking if you have ExtUtils::MakeMaker 0 ... Yes (7.44)
Building and testing Win32-0.53 ... cp Win32.pm blib\lib\Win32.pm
Running Mkbootstrap for Win32 ()
"C:\Users\cwhitener\_psperl\_perls\perl32w64int-5.30.0.1\perl\bin\perl.exe" -MExtUtils::Command -e chmod -- 644 "Win32.bs"
"C:\Users\cwhitener\_psperl\_perls\perl32w64int-5.30.0.1\perl\bin\perl.exe" -MExtUtils::Command::MM -e cp_nonempty -- Win32.bs blib\arch\auto\Win32\Win32.bs 644
"C:\Users\cwhitener\_psperl\_perls\perl32w64int-5.30.0.1\perl\bin\perl.exe" "C:\Users\cwhitener\_psperl\_perls\perl32w64int-5.30.0.1\perl\lib\ExtUtils/xsubpp"  -typemap C:\Users\cwhitener\_psperl\_perls\perl32w64int-5.30.0.1\perl\lib\ExtUtils\typemap  Win32.xs > Win32.xsc
"C:\Users\cwhitener\_psperl\_perls\perl32w64int-5.30.0.1\perl\bin\perl.exe" -MExtUtils::Command -e mv -- Win32.xsc Win32.c
gcc -c   -s -O2 -DWIN32 -D__USE_MINGW_ANSI_STDIO -DPERL_TEXTMODE_SCRIPTS -DPERL_IMPLICIT_CONTEXT -DPERL_IMPLICIT_SYS -DUSE_PERLIO -fwrapv -fno-strict-aliasing -mms-bitfields -s -O2   -DVERSION=\"0.53\" -DXS_VERSION=\"0.53\"  "-IC:\Users\cwhitener\_psperl\_perls\perl32w64int-5.30.0.1\perl\lib\CORE"   Win32.c
"C:\Users\cwhitener\_psperl\_perls\perl32w64int-5.30.0.1\perl\bin\perl.exe" -MExtUtils::Mksymlists \
     -e "Mksymlists('NAME'=>\"Win32\", 'DLBASE' => 'Win32', 'DL_FUNCS' => {  }, 'FUNCLIST' => [], 'IMPORTS' => {  }, 'DL_VARS' => []);"
g++.exe Win32.def -o blib\arch\auto\Win32\Win32.xs.dll -mdll -s -L"C:\Users\cwhitener\_psperl\_perls\perl32w64int-5.30.0.1\perl\lib\CORE" -L"C:\Users\cwhitener\_psperl\_perls\perl32w64int-5.30.0.1\c\lib" Win32.o   "C:\Users\cwhitener\_psperl\_perls\perl32w64int-5.30.0.1\perl\lib\CORE\libperl530.a" -lmoldname -lkernel32 -luser32 -lgdi32 -lwinspool -lcomdlg32 -ladvapi32 -lshell32 -lole32 -loleaut32 -lnetapi32 -luuid -lws2_32 -lmpr -lwinmm -lversion -lodbc32 -lodbccp32 -lcomctl32 -Wl,--enable-auto-image-base
"C:\Users\cwhitener\_psperl\_perls\perl32w64int-5.30.0.1\perl\bin\perl.exe" -MExtUtils::Command -e chmod -- 755 blib\arch\auto\Win32\Win32.xs.dll
"C:\Users\cwhitener\_psperl\_perls\perl32w64int-5.30.0.1\perl\bin\perl.exe" -MExtUtils::Command::MM -e cp_nonempty -- Win32.bs blib\arch\auto\Win32\Win32.bs 644
"C:\Users\cwhitener\_psperl\_perls\perl32w64int-5.30.0.1\perl\bin\perl.exe" "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(0, 'blib\lib', 'blib\arch')" t/*.t
t/CodePage.t .................. ok
t/CreateFile.t ................ ok
t/ExpandEnvironmentStrings.t .. ok
t/GetCurrentThreadId.t ........ ok
t/GetFileVersion.t ............ skipped: Only ActivePerl seems to set the perl.exe fileversion
t/GetFolderPath.t ............. ok
t/GetFullPathName.t ........... ok
t/GetLongPathName.t ........... ok
t/GetOSName.t ................. ok
t/GetOSVersion.t .............. ok
t/GetShortPathName.t .......... 1/5 # Test 2 got: "Long Path 11684" (t/GetShortPathName.t at line 15)
#   Expected: "(?^:^\\S{1,8}(\\.\\S{1,3})?$)"
#  t/GetShortPathName.t line 15 is: ok($short, qr/^\S{1,8}(\.\S{1,3})?$/);
t/GetShortPathName.t .......... Failed 1/5 subtests
t/GuidGen.t ................... ok
t/Names.t ..................... ok
t/Unicode.t ................... # Failed test 1 in t/Unicode.t at line 43
t/Unicode.t ................... 1/12 #  t/Unicode.t line 43 is: ok(-d Win32::GetANSIPathName($dir));
# Failed test 2 in t/Unicode.t at line 47
#  t/Unicode.t line 47 is: ok(-f Win32::GetANSIPathName($file));
# Failed test 3 in t/Unicode.t at line 50
#  t/Unicode.t line 50 is: ok(opendir(my $dh, Win32::GetANSIPathName($dir)));
# Failed test 5 in t/Unicode.t at line 67
#  t/Unicode.t line 67 is: ok(Win32::SetCwd($dir));
# Test 8 got: "C:\\Users\\cwhitener\\_psperl\\_perls\\perl32w64int-5.30.0.1\\data\\.cpanm\\work\\1584363053.9400\\Win32-0.53" (t/Unicode.t at line 78)
#   Expected: "C:\\Users\\cwhitener\\_psperl\\_perls\\perl32w64int-5.30.0.1\\data\\.cpanm\\work\\1584363053.9400\\Win32-0.53\\Foo \x{394}\x{419} Bar \x{5E7}\x{645} Baz"
#  t/Unicode.t line 78 is: ok(Win32::GetLongPathName($w32dir), $long);
# Test 9 got: "C:\\Users\\cwhitener\\_psperl\\_perls\\perl32w64int-5.30.0.1\\data\\.cpanm\\work\\1584363053.9400\\Win32-0.53" (t/Unicode.t at line 86)
#   Expected: "C:\\Users\\cwhitener\\_psperl\\_perls\\perl32w64int-5.30.0.1\\data\\.cpanm\\work\\1584363053.9400\\Win32-0.53\\Foo \x{394}\x{419} Bar \x{5E7}\x{645} Baz"
#  t/Unicode.t line 86 is: ok(Win32::GetLongPathName($subdir), $long);
# Failed test 10 in t/Unicode.t at line 89
#  t/Unicode.t line 89 is: ok(chdir(Win32::GetANSIPathName($dir)));
# Test 11 got: "C:\\Users\\cwhitener\\_psperl\\_perls\\perl32w64int-5.30.0.1\\data\\.cpanm\\work\\1584363053.9400\\Win32-0.53" (t/Unicode.t at line 90)
#    Expected: "C:\\Users\\cwhitener\\_psperl\\_perls\\perl32w64int-5.30.0.1\\data\\.cpanm\\work\\1584363053.9400\\Win32-0.53\\Foo \x{394}\x{419} Bar \x{5E7}\x{645} Baz"
#  t/Unicode.t line 90 is: ok(Win32::GetLongPathName(Win32::GetCwd()), $long);
t/Unicode.t ................... Failed 9/12 subtests

Test Summary Report
-------------------
t/GetShortPathName.t        (Wstat: 0 Tests: 5 Failed: 1)
  Failed test:  2
t/Unicode.t                 (Wstat: 0 Tests: 11 Failed: 8)
  Failed tests:  1-3, 5, 8-11
  Parse errors: Bad plan.  You planned 12 tests but ran 11.
Files=14, Tests=638,  3 wallclock secs ( 0.06 usr +  0.05 sys =  0.11 CPU)
Result: FAIL
Failed 2/14 test programs. 9/638 subtests failed.
gmake: *** [Makefile:958: test_dynamic] Error 255
FAIL
! Testing Win32 failed. See C:\Users\cwhitener\_psperl\_perls\perl32w64int-5.30.0.1\data\.cpanm\work\1584363053.9400\build.log for details. Retry with --force to force install it.
PS C:\Users\cwhitener>
```

**Comment by gordon-fish** [comment]
@genio ``t/Unicode.t line 47 is: ok(-f Win32::GetANSIPathName($file));`` and ``t/Unicode.t line 50 is: ok(opendir(my $dh, Win32::GetANSIPathName($dir)));`` suggest to me that there is a permissions problem.

I did some checking and found that if I remove the ``Create Folders / Append Data`` permission on the ``Win32-0.53`` directory, I get the same errors from ``Unicode.t``, complete with the cut-off paths, and if I restore it again, all tests ``PASS`` again, so check the permissions on that directory.

(I tested this under 10, 7, and XP and got the same results all around, if anyone is curious.)

**Comment by genio** [comment]
@gordon-fish I went to my top level `C:\Users\cwhitener\_psperl\` directory and told it to remove the read-only flag recursively (takes a while with over 3GB of stuff in there -multiple Perls). Once it was complete, I then went to the working directories:

* `C:\Users\cwhitener\_psperl\_perls\perl32w64int-5.30.0.1\`
* `C:\Users\cwhitener\_psperl\_perls\perl32w64int-5.30.0.1\data\`
* `C:\Users\cwhitener\_psperl\_perls\perl32w64int-5.30.0.1\data\.cpanm\work\`

And checked those as well. There are no read-only flags set. I then re-ran `cpanm -v Win32` to have it fail again in the same way.

No matter how many times you clear that read-only flag, it gets reset because `C:\Users` is marked read-only by default.

**Comment by genio** [comment]
I have full control on that directory and all the way up:

```log
PS C:\Users\cwhitener> cpanm -v Win32
cpanm (App::cpanminus) 1.7044 on perl 5.028002 built for MSWin32-x64-multi-thread
Work directory is C:\Users\cwhitener\_psperl\_perls\perl64-5.28.2.1\data/.cpanm/work/1584631092.856
You have make C:\Users\cwhitener\_psperl\_perls\perl64-5.28.2.1\c\bin\gmake.exe
You have LWP 6.38
Falling back to Archive::Tar 2.36
Searching Win32 () on cpanmetadb ...
--> Working on Win32
Fetching http://www.cpan.org/authors/id/J/JD/JDB/Win32-0.53.tar.gz ... OK
Unpacking Win32-0.53.tar.gz
Entering Win32-0.53
Checking configure dependencies from META.yml
Running Makefile.PL
Configuring Win32-0.53 ... Checking if your kit is complete...
Looks good
Generating a gmake-style Makefile
Writing Makefile for Win32
Writing MYMETA.yml and MYMETA.json
OK
Checking dependencies from MYMETA.json ...
Checking if you have ExtUtils::MakeMaker 0 ... Yes (7.44)
Building and testing Win32-0.53 ... cp Win32.pm blib\lib\Win32.pm
Running Mkbootstrap for Win32 ()
"C:\Users\cwhitener\_psperl\_perls\perl64-5.28.2.1\perl\bin\perl.exe" -MExtUtils::Command -e chmod -- 644 "Win32.bs"
"C:\Users\cwhitener\_psperl\_perls\perl64-5.28.2.1\perl\bin\perl.exe" -MExtUtils::Command::MM -e cp_nonempty -- Win32.bs blib\arch\auto\Win32\Win32.bs 644
"C:\Users\cwhitener\_psperl\_perls\perl64-5.28.2.1\perl\bin\perl.exe" "C:\Users\cwhitener\_psperl\_perls\perl64-5.28.2.1\perl\lib\ExtUtils/xsubpp"  -typemap C:\Users\cwhitener\_psperl\_perls\perl64-5.28.2.1\perl\lib\ExtUtils\typemap  Win32.xs > Win32.xsc
"C:\Users\cwhitener\_psperl\_perls\perl64-5.28.2.1\perl\bin\perl.exe" -MExtUtils::Command -e mv -- Win32.xsc Win32.c
gcc -c   -s -O2 -DWIN32 -DWIN64 -DCONSERVATIVE -D__USE_MINGW_ANSI_STDIO -DPERL_TEXTMODE_SCRIPTS -DPERL_IMPLICIT_CONTEXT -DPERL_IMPLICIT_SYS -DUSE_PERLIO -fwrapv -fno-strict-aliasing -mms-bitfields -s -O2   -DVERSION=\"0.53\" -DXS_VERSION=\"0.53\"  "-IC:\Users\cwhitener\_psperl\_perls\perl64-5.28.2.1\perl\lib\CORE"   Win32.c
"C:\Users\cwhitener\_psperl\_perls\perl64-5.28.2.1\perl\bin\perl.exe" -MExtUtils::Mksymlists \
     -e "Mksymlists('NAME'=>\"Win32\", 'DLBASE' => 'Win32', 'DL_FUNCS' => {  }, 'FUNCLIST' => [], 'IMPORTS' => {  }, 'DL_VARS' => []);"
g++.exe Win32.def -o blib\arch\auto\Win32\Win32.xs.dll -mdll -s -L"C:\Users\cwhitener\_psperl\_perls\perl64-5.28.2.1\perl\lib\CORE" -L"C:\Users\cwhitener\_psperl\_perls\perl64-5.28.2.1\c\lib" Win32.o   "C:\Users\cwhitener\_psperl\_perls\perl64-5.28.2.1\perl\lib\CORE\libperl528.a" -lmoldname -lkernel32 -luser32 -lgdi32 -lwinspool -lcomdlg32 -ladvapi32 -lshell32 -lole32 -loleaut32 -lnetapi32 -luuid -lws2_32 -lmpr -lwinmm -lversion -lodbc32 -lodbccp32 -lcomctl32 -Wl,--enable-auto-image-base
"C:\Users\cwhitener\_psperl\_perls\perl64-5.28.2.1\perl\bin\perl.exe" -MExtUtils::Command -e chmod -- 755 blib\arch\auto\Win32\Win32.xs.dll
"C:\Users\cwhitener\_psperl\_perls\perl64-5.28.2.1\perl\bin\perl.exe" -MExtUtils::Command::MM -e cp_nonempty -- Win32.bs blib\arch\auto\Win32\Win32.bs 644
"C:\Users\cwhitener\_psperl\_perls\perl64-5.28.2.1\perl\bin\perl.exe" "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(0, 'blib\lib', 'blib\arch')" t/*.t
t/CodePage.t .................. ok
t/CreateFile.t ................ ok
t/ExpandEnvironmentStrings.t .. ok
t/GetCurrentThreadId.t ........ ok
t/GetFileVersion.t ............ skipped: Only ActivePerl seems to set the perl.exe fileversion
t/GetFolderPath.t ............. ok
t/GetFullPathName.t ........... ok
t/GetLongPathName.t ........... ok
t/GetOSName.t ................. ok
t/GetOSVersion.t .............. ok
t/GetShortPathName.t .......... 1/5 # Test 2 got: "Long Path 2976" (t/GetShortPathName.t at line 15)
#   Expected: "(?^:^\\S{1,8}(\\.\\S{1,3})?$)"
#  t/GetShortPathName.t line 15 is: ok($short, qr/^\S{1,8}(\.\S{1,3})?$/);
t/GetShortPathName.t .......... Failed 1/5 subtests
t/GuidGen.t ................... ok
t/Names.t ..................... ok
t/Unicode.t ................... # Failed test 1 in t/Unicode.t at line 43
t/Unicode.t ................... 1/12 #  t/Unicode.t line 43 is: ok(-d Win32::GetANSIPathName($dir));
# Failed test 2 in t/Unicode.t at line 47
#  t/Unicode.t line 47 is: ok(-f Win32::GetANSIPathName($file));
# Failed test 3 in t/Unicode.t at line 50
#  t/Unicode.t line 50 is: ok(opendir(my $dh, Win32::GetANSIPathName($dir)));
# Failed test 5 in t/Unicode.t at line 67
#  t/Unicode.t line 67 is: ok(Win32::SetCwd($dir));
# Test 8 got: "C:\\Users\\cwhitener\\_psperl\\_perls\\perl64-5.28.2.1\\data\\.cpanm\\work\\1584631092.856\\Win32-0.53" (t/Unicode.t at line 78)
#   Expected: "C:\\Users\\cwhitener\\_psperl\\_perls\\perl64-5.28.2.1\\data\\.cpanm\\work\\1584631092.856\\Win32-0.53\\Foo \x{394}\x{419} Bar \x{5E7}\x{645} Baz"
#  t/Unicode.t line 78 is: ok(Win32::GetLongPathName($w32dir), $long);
# Test 9 got: "C:\\Users\\cwhitener\\_psperl\\_perls\\perl64-5.28.2.1\\data\\.cpanm\\work\\1584631092.856\\Win32-0.53" (t/Unicode.t at line 86)
#   Expected: "C:\\Users\\cwhitener\\_psperl\\_perls\\perl64-5.28.2.1\\data\\.cpanm\\work\\1584631092.856\\Win32-0.53\\Foo \x{394}\x{419} Bar \x{5E7}\x{645} Baz"
#  t/Unicode.t line 86 is: ok(Win32::GetLongPathName($subdir), $long);
# Failed test 10 in t/Unicode.t at line 89
#  t/Unicode.t line 89 is: ok(chdir(Win32::GetANSIPathName($dir)));
# Test 11 got: "C:\\Users\\cwhitener\\_psperl\\_perls\\perl64-5.28.2.1\\data\\.cpanm\\work\\1584631092.856\\Win32-0.53" (t/Unicode.t at line 90)
#    Expected: "C:\\Users\\cwhitener\\_psperl\\_perls\\perl64-5.28.2.1\\data\\.cpanm\\work\\1584631092.856\\Win32-0.53\\Foo \x{394}\x{419} Bar \x{5E7}\x{645} Baz"
#  t/Unicode.t line 90 is: ok(Win32::GetLongPathName(Win32::GetCwd()), $long);
t/Unicode.t ................... Failed 9/12 subtests

Test Summary Report
-------------------
t/GetShortPathName.t        (Wstat: 0 Tests: 5 Failed: 1)
  Failed test:  2
t/Unicode.t                 (Wstat: 0 Tests: 11 Failed: 8)
  Failed tests:  1-3, 5, 8-11
  Parse errors: Bad plan.  You planned 12 tests but ran 11.
Files=14, Tests=638,  2 wallclock secs ( 0.09 usr +  0.03 sys =  0.13 CPU)
Result: FAIL
Failed 2/14 test programs. 9/638 subtests failed.
gmake: *** [Makefile:957: test_dynamic] Error 255
FAIL
! Installing Win32 failed. See C:\Users\cwhitener\_psperl\_perls\perl64-5.28.2.1\data\.cpanm\work\1584631092.856\build.log for details. Retry with --force to force install it.
PS C:\Users\cwhitener>
```
I've included some images of what the permissions look like for that directory as well as a directory listing so you can see the folder got created:

![cpanm_show_dir](https://user-images.githubusercontent.com/109551/77083906-9874dc00-69d4-11ea-804b-e16ed5a21c65.png)

![cpanm_show_dir_perms](https://user-images.githubusercontent.com/109551/77083908-9874dc00-69d4-11ea-8f84-9f46f2631738.png)

**Comment by genio** [comment]
Did the same by running in cmd.exe as administrator:

```log
----------------------------------------------
 Welcome to Strawberry Perl Portable Edition!
 * URL - http://www.strawberryperl.com/
 * see README.TXT for more info
----------------------------------------------
Perl executable: C:\Users\cwhitener\_psperl\_perls\perl64-5.28.2.1\perl\bin\perl.exe
Perl version   : 5.28.2 / MSWin32-x64-multi-thread

C:\WINDOWS\system32>cpanm --test-only -v Win32
cpanm (App::cpanminus) 1.7044 on perl 5.028002 built for MSWin32-x64-multi-thread
Work directory is C:\Users\cwhitener\_psperl\_perls\perl64-5.28.2.1\data/.cpanm/work/1584632667.13812
You have make C:\Users\cwhitener\_psperl\_perls\perl64-5.28.2.1\c\bin\gmake.exe
You have LWP 6.38
Falling back to Archive::Tar 2.32
Searching Win32 () on cpanmetadb ...
--> Working on Win32
Fetching http://www.cpan.org/authors/id/J/JD/JDB/Win32-0.53.tar.gz ... OK
Unpacking Win32-0.53.tar.gz
Entering Win32-0.53
Checking configure dependencies from META.yml
Running Makefile.PL
Configuring Win32-0.53 ... Checking if your kit is complete...
Looks good
Generating a gmake-style Makefile
Writing Makefile for Win32
Writing MYMETA.yml and MYMETA.json
OK
Checking dependencies from MYMETA.json ...
Checking if you have ExtUtils::MakeMaker 0 ... Yes (7.36)
Building and testing Win32-0.53 ... cp Win32.pm blib\lib\Win32.pm
Running Mkbootstrap for Win32 ()
"C:\Users\cwhitener\_psperl\_perls\perl64-5.28.2.1\perl\bin\perl.exe" -MExtUtils::Command -e chmod -- 644 "Win32.bs"
"C:\Users\cwhitener\_psperl\_perls\perl64-5.28.2.1\perl\bin\perl.exe" -MExtUtils::Command::MM -e cp_nonempty -- Win32.bs blib\arch\auto\Win32\Win32.bs 644
"C:\Users\cwhitener\_psperl\_perls\perl64-5.28.2.1\perl\bin\perl.exe" "C:\Users\cwhitener\_psperl\_perls\perl64-5.28.2.1\perl\lib\ExtUtils/xsubpp"  -typemap C:\Users\cwhitener\_psperl\_perls\perl64-5.28.2.1\perl\lib\ExtUtils\typemap  Win32.xs > Win32.xsc
"C:\Users\cwhitener\_psperl\_perls\perl64-5.28.2.1\perl\bin\perl.exe" -MExtUtils::Command -e mv -- Win32.xsc Win32.c
gcc -c   -s -O2 -DWIN32 -DWIN64 -DCONSERVATIVE -D__USE_MINGW_ANSI_STDIO -DPERL_TEXTMODE_SCRIPTS -DPERL_IMPLICIT_CONTEXT -DPERL_IMPLICIT_SYS -DUSE_PERLIO -fwrapv -fno-strict-aliasing -mms-bitfields -s -O2   -DVERSION=\"0.53\" -DXS_VERSION=\"0.53\"  "-IC:\Users\cwhitener\_psperl\_perls\perl64-5.28.2.1\perl\lib\CORE"   Win32.c
"C:\Users\cwhitener\_psperl\_perls\perl64-5.28.2.1\perl\bin\perl.exe" -MExtUtils::Mksymlists \
     -e "Mksymlists('NAME'=>\"Win32\", 'DLBASE' => 'Win32', 'DL_FUNCS' => {  }, 'FUNCLIST' => [], 'IMPORTS' => {  }, 'DL_VARS' => []);"
g++.exe Win32.def -o blib\arch\auto\Win32\Win32.xs.dll -mdll -s -L"C:\Users\cwhitener\_psperl\_perls\perl64-5.28.2.1\perl\lib\CORE" -L"C:\Users\cwhitener\_psperl\_perls\perl64-5.28.2.1\c\lib" Win32.o   "C:\Users\cwhitener\_psperl\_perls\perl64-5.28.2.1\perl\lib\CORE\libperl528.a" -lmoldname -lkernel32 -luser32 -lgdi32 -lwinspool -lcomdlg32 -ladvapi32 -lshell32 -lole32 -loleaut32 -lnetapi32 -luuid -lws2_32 -lmpr -lwinmm -lversion -lodbc32 -lodbccp32 -lcomctl32 -Wl,--enable-auto-image-base
"C:\Users\cwhitener\_psperl\_perls\perl64-5.28.2.1\perl\bin\perl.exe" -MExtUtils::Command -e chmod -- 755 blib\arch\auto\Win32\Win32.xs.dll
"C:\Users\cwhitener\_psperl\_perls\perl64-5.28.2.1\perl\bin\perl.exe" -MExtUtils::Command::MM -e cp_nonempty -- Win32.bs blib\arch\auto\Win32\Win32.bs 644
"C:\Users\cwhitener\_psperl\_perls\perl64-5.28.2.1\perl\bin\perl.exe" "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(0, 'blib\lib', 'blib\arch')" t/*.t
t/CodePage.t .................. ok
t/CreateFile.t ................ ok
t/ExpandEnvironmentStrings.t .. ok
t/GetCurrentThreadId.t ........ ok
t/GetFileVersion.t ............ skipped: Only ActivePerl seems to set the perl.exe fileversion
t/GetFolderPath.t ............. ok
t/GetFullPathName.t ........... ok
t/GetLongPathName.t ........... ok
t/GetOSName.t ................. ok
t/GetOSVersion.t .............. ok
t/GetShortPathName.t .......... 1/5 # Test 2 got: "Long Path 13028" (t/GetShortPathName.t at line 15)
#   Expected: "(?^:^\\S{1,8}(\\.\\S{1,3})?$)"
#  t/GetShortPathName.t line 15 is: ok($short, qr/^\S{1,8}(\.\S{1,3})?$/);
t/GetShortPathName.t .......... Failed 1/5 subtests
t/GuidGen.t ................... ok
t/Names.t ..................... ok
t/Unicode.t ................... # Failed test 1 in t/Unicode.t at line 43
#  t/Unicode.t line 43 is: ok(-d Win32::GetANSIPathName($dir));
t/Unicode.t ................... 1/12 # Failed test 2 in t/Unicode.t at line 47
#  t/Unicode.t line 47 is: ok(-f Win32::GetANSIPathName($file));
# Failed test 3 in t/Unicode.t at line 50
#  t/Unicode.t line 50 is: ok(opendir(my $dh, Win32::GetANSIPathName($dir)));
# Failed test 5 in t/Unicode.t at line 67
#  t/Unicode.t line 67 is: ok(Win32::SetCwd($dir));
# Test 8 got: "C:\\Users\\cwhitener\\_psperl\\_perls\\perl64-5.28.2.1\\data\\.cpanm\\work\\1584632667.13812\\Win32-0.53" (t/Unicode.t at line 78)
#   Expected: "C:\\Users\\cwhitener\\_psperl\\_perls\\perl64-5.28.2.1\\data\\.cpanm\\work\\1584632667.13812\\Win32-0.53\\Foo \x{394}\x{419} Bar \x{5E7}\x{645} Baz"
#  t/Unicode.t line 78 is: ok(Win32::GetLongPathName($w32dir), $long);
# Test 9 got: "C:\\Users\\cwhitener\\_psperl\\_perls\\perl64-5.28.2.1\\data\\.cpanm\\work\\1584632667.13812\\Win32-0.53" (t/Unicode.t at line 86)
#   Expected: "C:\\Users\\cwhitener\\_psperl\\_perls\\perl64-5.28.2.1\\data\\.cpanm\\work\\1584632667.13812\\Win32-0.53\\Foo \x{394}\x{419} Bar \x{5E7}\x{645} Baz"
#  t/Unicode.t line 86 is: ok(Win32::GetLongPathName($subdir), $long);
# Failed test 10 in t/Unicode.t at line 89
#  t/Unicode.t line 89 is: ok(chdir(Win32::GetANSIPathName($dir)));
# Test 11 got: "C:\\Users\\cwhitener\\_psperl\\_perls\\perl64-5.28.2.1\\data\\.cpanm\\work\\1584632667.13812\\Win32-0.53" (t/Unicode.t at line 90)
#    Expected: "C:\\Users\\cwhitener\\_psperl\\_perls\\perl64-5.28.2.1\\data\\.cpanm\\work\\1584632667.13812\\Win32-0.53\\Foo \x{394}\x{419} Bar \x{5E7}\x{645} Baz"
#  t/Unicode.t line 90 is: ok(Win32::GetLongPathName(Win32::GetCwd()), $long);
t/Unicode.t ................... Failed 9/12 subtests

Test Summary Report
-------------------
t/GetShortPathName.t        (Wstat: 0 Tests: 5 Failed: 1)
  Failed test:  2
t/Unicode.t                 (Wstat: 0 Tests: 11 Failed: 8)
  Failed tests:  1-3, 5, 8-11
  Parse errors: Bad plan.  You planned 12 tests but ran 11.
Files=14, Tests=638,  1 wallclock secs ( 0.12 usr +  0.01 sys =  0.14 CPU)
Result: FAIL
Failed 2/14 test programs. 9/638 subtests failed.
gmake: *** [Makefile:956: test_dynamic] Error 255
FAIL
! Testing Win32 failed. See C:\Users\cwhitener\_psperl\_perls\perl64-5.28.2.1\data\.cpanm\work\1584632667.13812\build.log for details. Retry with --force to force install it.

C:\WINDOWS\system32>
```

**Comment by xenu** [comment]
See #23

**Comment by jandubois** [comment]
#23 has been merged and is included in Win32-0.54 on CPAN.

Please reopen if you still run into problems with 0.54.
